sdk/clientcore/core/src/main/java/io/clientcore/core/util/auth/AuthUtils.java

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.auth;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.clientcore.core.util.auth.AuthScheme.BASIC;
+
+/**
+ * Utility class for handling various HTTP authentication headers.
+ */
+public class AuthUtils {
```

Comment:
Does this class need to be in the public API?

Reply:
I think I was using it in okhttp3 lib too, I will check if we should rather duplicate there.